mantidproject/mantid
# Fix Instrument Window segfaulting when the instrument changes

Changing a workspace's instrument while the Instrument Window is open would invalidate some pointers pointing to an `InstrumentActor` object. This could lead to segmentation fault later when user tried to use the Instrument Window. This PR fixes the segfault by forcing a single point access to the `InstrumentActor` object.

**To test:**

Try to reproduce the segfault first on master, then repeat on this branch to check that the segfault is really gone.
1. Use for instance [`LoadEmptyInstrument`](http://docs.mantidproject.org/nightly/algorithms/LoadEmptyInstrument-v1.html) to get a workspace with instrument.
2. Open Instrument Window for the workspace.
3. Choose the 'Pick' tab.
4. Execute `LoadEmptyInstrument` again with the same OutputWorkspace.
5. Try pointing the instrument's components in the Instrument Window. There is a change that MantidPlot segfaults if these changes are not applied. You might need to go back to step 4.


Fixes #16957.

*Release notes were updated accordingly.*

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
